Improve plugin Code Coverage

prudhvigodithi opened the pull request:
### Description
This is the 1st PR as part of this [issue]( https://github.com/opensearch-project/job-scheduler/issues/612) to improve the repo code coverage gradually.
 
### Issues Resolved
Part of https://github.com/opensearch-project/job-scheduler/issues/612
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

prudhvigodithi commented:
FYI, I allowed the tests to run few times since yesterday and I get the following error.
```
{'detail': ErrorDetail(string='Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected available in 2831 seconds.', code='throttled')}
```
Let me add a Codecov token and update the test workflow to use the token for uploading the test report.
@bbarani @cwperks

prudhvigodithi commented:
Works now https://app.codecov.io/gh/opensearch-project/job-scheduler/pull/616 and I see the percentage is now 37.27% (from 29.17%).
<img width="1375" alt="Screenshot 2024-04-29 at 11 02 41 AM" src="https://github.com/opensearch-project/job-scheduler/assets/28733332/b211a671-a928-4724-a53a-2a73e5614b31">
@cwperks @joshpalis can you please check?
@bbarani

cwperks commented on src/test/java/org/opensearch/jobscheduler/JobSchedulerPluginTests.java:
```diff
@@ -0,0 +1,169 @@
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .build();
+        indexSettings = IndexSettingsModule.newIndexSettings(".opensearch-job-scheduler-job-details", settings);
```
Import this from JobDetailsService?

prudhvigodithi replied:
Let me check this.

prudhvigodithi replied:
Just pushed in my latest commit.
Thanks

cwperks commented on src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java:
```diff
@@ -137,7 +137,7 @@ void updateIndexToJobProviders(String documentId, JobDetails jobDetails) {
      * @param jobDetails the jobDetails to register
      */
-    void updateIndexToJobDetails(String documentId, JobDetails jobDetails) {
+    public void updateIndexToJobDetails(String documentId, JobDetails jobDetails) {
```
Is this change necessary?

cwperks commented on src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java:
```diff
@@ -69,14 +69,13 @@ public class JobSchedulerPlugin extends Plugin implements ActionPlugin, Extensib
     public static final String JS_BASE_URI = "/_plugins/_job_scheduler";
 
     private static final Logger log = LogManager.getLogger(JobSchedulerPlugin.class);
+    JobSweeper sweeper;
```
Are the changes to the access modifiers necessary here too?

cwperks reviewed (commented):
Took a first pass. Thank you for adding missing test! This change looks good to me, just left some minor comments.

prudhvigodithi commented:
Thanks @cwperks for reviewing the PR, initially I have to update the access modifiers to make sure those variables/methods are accessible for testing, I have reverted back and replaced to get the getters, can you review again?
